src/Bicep.Local.Deploy/LocalDeploymentEngineHost.cs

```diff
@@ -140,19 +148,29 @@ protected override async Task<string> TryReadAsString(HttpContent content, bool
         }
     }
 
-    public override async Task<HttpResponseMessage> CallExtensibilityHost(
+    public override async Task<HttpResponseMessage> CallExtensibilityHostV2(
         HttpMethod requestMethod,
         Uri requestUri,
-        ExtensibilityOperationRequest request,
+        HttpContent content,
         AuthenticationToken extensibilityHostToken,
         CancellationToken cancellationToken)
     {
-        var response = await extensibilityHandler.CallExtensibilityHost(requestUri.Segments[^1], request, cancellationToken);
+        var extensionName = requestUri.Segments[^4].TrimEnd('/');
+        var extensionVersion = requestUri.Segments[^3].TrimEnd('/');
+        var method = requestUri.Segments[^1].TrimEnd('/');
+
+        var extensionInfo = new ExtensionInfo(extensionName, extensionVersion, method);
+        var extensibilityResponse = await extensibilityHandler.CallExtensibilityHost(extensionInfo, content, cancellationToken);
 
-        return new HttpResponseMessage(HttpStatusCode.OK)
+        var response = new HttpResponseMessage(HttpStatusCode.OK)
         {
-            Content = new StringContent(response.ToJson()),
+            Content = new StringContent(extensibilityResponse.ToJson(), encoding: Encoding.UTF8, mediaType: "application/json")
         };
+
+        response.Headers.Add("Location", "local");
+        response.Headers.Add("Version", extensionVersion);
```

Comment:
What are these headers used for? Would you mind adding a code comment to explain?

Reply:
@shenglol are we going to relax echo these header values in deployment engine? If so, these headers are just temporal additions and can add a comment but if is permanent, are these values ok?

To give you some context Anthony, deployment engine performs validations on header values and expects these two to be present, but it might not apply to local extensibility.

Reply:
Yeah, I'll send a PR to relax the validation for the headers. We can remove them from LocalDeploymentEngineHost later.